ui/views.py

```diff
@@ -240,7 +241,12 @@ def extra_context(self):
 @login_required
 def export(request, resource_id):
     """Dump LearningResource as XML"""
-    return HttpResponse(
-        get_resource(resource_id, request.user.id).content_xml,
-        content_type='text/xml'
-    )
+    try:
+        return HttpResponse(
+            get_resource(resource_id, request.user.id).content_xml,
+            content_type='text/xml'
+        )
+    except NotFound:
+        raise Http404()
+    except PermissionDenied:
+        return HttpResponseForbidden("unauthorized")
```

Comment:
On second look, I don't think using Django's `PermissionDenied` would work. It's not a response.